### PR TITLE
Introduce `Redacted<T>` module

### PR DESCRIPTION
**Pull request by KhraksMamtsov**
## Type

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

- Introduce `Redacted<T>` module - as mentioned here https://github.com/Effect-TS/effect/pull/2810
- `Secret extends Redacted<string>`
- Have added two related schemas and remove `Secret` and `SecretFromSelf`
- Have added tests
- Have Changed `Secret` usage on `Redacted<string>` in packages with version `0.*.*`

## Related

- Related Issue #
- Closes #

### CODE REVIEW

**Comment by mikearnaldi** [comment]
Accessing a value after wipe should probably throw instead of returning undefined

**Comment by mikearnaldi** [comment]
Wondering if we should go for Hidden or Redacted as a name, I sort of like Redacted more because it is in line with what's shown in logs and has precedents in other languages like https://github.com/polentino/redacted

**Comment by KhraksMamtsov** [comment]
Understood, I will rename it in the near future

**Review by tim-smart — CHANGES_REQUESTED** [review]

**Comment by mikearnaldi** [comment]
Re failing test this is due to missing re-export from index, you can add that by running `pnpm codegen`

File: packages/effect/src/internal/redacted.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  }
+}
+
+/** @internal */
+export const isRedacted = (u: unknown): u is Redacted.Redacted<unknown> => hasProperty(u, RedactedTypeId)
+
+/** @internal */
+export const make = <T>(value: T): Redacted.Redacted<T> => {
+  const redacted = Object.create(proto)
+  Object.defineProperty(redacted, "toString", {
```
[the remark this answers is not in the document]
Comment (KhraksMamtsov) [reply packages/effect/src/internal/redacted.ts]: Done

File: packages/effect/src/Redacted.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+ * @since 3.1.3
+ * @category symbols
+ */
+export type RedactedTypeId = typeof RedactedTypeId
+
+/**
+ * @since 3.1.3
+ * @category models
+ */
+export interface Redacted<out A> extends Redacted.Variance<A>, Equal.Equal, Pipeable {
```
[the remark this answers is not in the document]
Comment (KhraksMamtsov) [reply packages/effect/src/Redacted.ts]: I agree - I corrected it

File: packages/cli/src/Args.ts [hunk 3]
```diff
@@ -379,12 +379,12 @@ export const repeated: <A>(self: Args<A>) => Args<Array<A>> = InternalArgs.repea
...
  * Creates a text argument.
  *
- * Can optionally provide a custom argument name (defaults to `"secret"`).
+ * Can optionally provide a custom argument name (defaults to `"redacted"`).
  *
  * @since 1.0.0
  * @category constructors
  */
-export const secret: (config?: Args.BaseArgsConfig) => Args<Secret> = InternalArgs.secret
+export const hidden: (config?: Args.BaseArgsConfig) => Args<Redacted<string>> = InternalArgs.hidden
```
[the remark this answers is not in the document]
Comment (KhraksMamtsov) [reply packages/cli/src/Args.ts]: Ok, then I will keep `Secret`&`SecretFromSelf` schemas for `@effect/schema` as well

File: .changeset/slow-paws-check.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+"@effect/sql-pg": patch
+"@effect/vitest": patch
+"@effect/cli": patch
+"@effect/rpc": patch
+"@effect/sql": patch
+---
+
+Introduced `Redacted<T>` module - `Secret` generalization
+`Secret extends Redacted<string>`
+Added two related schemas `Redacted` and `RedactedFromSelf`
```
[the remark this answers is not in the document]
Comment (KhraksMamtsov) [reply .changeset/slow-paws-check.md:7]: done

**Comment by KhraksMamtsov** [comment]
It seems that I have corrected all the comments,
please point out the flaws if something is wrong

**Review by tim-smart — APPROVED** [review]

**Pull request merged**